magnum-zio/src/main/scala/com/augustnagro/magnum/magzio/Transactor.scala

```diff
@@ -1,36 +1,32 @@
 package com.augustnagro.magnum.magzio
 
-import com.augustnagro.magnum.{DbCon, DbTx, SqlException, SqlLogger}
+import com.augustnagro.magnum.{
+  DbCon,
+  DbTx,
+  SqlException,
+  SqlLogger,
+  TransactorOps
+}
 import zio.{Semaphore, Task, Trace, UIO, ULayer, ZIO, ZLayer}
 
 import java.sql.Connection
 import javax.sql.DataSource
 import scala.util.control.NonFatal
 
-class Transactor private (
-    dataSource: DataSource,
-    sqlLogger: SqlLogger,
-    connectionConfig: Connection => Unit,
-    semaphore: Option[Semaphore]
-):
+final case class Transactor(
+    private val dataSource: DataSource,
+    private val sqlLogger: SqlLogger,
+    private val connectionConfig: Connection => Unit,
+    private val semaphore: Option[Semaphore]
+) extends TransactorOps[Task]:
 
   def withSqlLogger(sqlLogger: SqlLogger): Transactor =
-    new Transactor(
-      dataSource,
-      sqlLogger,
-      connectionConfig,
-      semaphore
-    )
+    copy(sqlLogger = sqlLogger)
```

Comment:
+1

Reply:
This is a benefit of services as case classes.

Reply:
indeed but I still prefer it not to be a case class

Reply:
No worries - I can respect that.